tests/release/conftest.py

```diff
@@ -0,0 +1,158 @@
+def output_dir() -> Path:
+    # TODO: [MIC-5522] define correct output dir
+    # output_dir = os.environ.get("PSP_TEST_OUTPUT_DIR")
+    output_dir_name = (
+        "/mnt/team/simulation_science/priv/engineering/pseudopeople_release_testing"
+    )
+    if not output_dir_name:
+        raise ValueError("PSP_TEST_OUTPUT_DIR environment variable not set")
+    output_dir = Path(output_dir_name) / f"{time.strftime('%Y%m%d_%H%M%S')}"
+    output_dir.mkdir(parents=True, exist_ok=False)
+    return output_dir.resolve()
+
+
+@pytest.fixture(scope="session")
+def dataset(output_dir: Path, request: pytest.FixtureRequest) -> pd.DataFrame:
+    dataset_name, dataset_func, source, engine, state, year = _parse_dataset_params(request)
+
+    if dataset_func == generate_social_security:
+        return profile_data_generation(output_dir)(dataset_func)(
+            source=source, year=year, engine=engine
+        )
+    else:
+        return profile_data_generation(output_dir)(dataset_func)(
+            source=source, year=year, state=state, engine=engine
+        )
+
+
+####################
+# Helper Functions #
+####################
+def profile_data_generation(output_dir: Path) -> Callable[..., Callable[..., pd.DataFrame]]:
+    """Decorator to profile a function's time and memory usage."""
+    # TODO: [MIC-5522] properly setup profiling
+    def decorator(func: Callable[..., pd.DataFrame]) -> Callable[..., pd.DataFrame]:
+        @functools.wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> pd.DataFrame:
+            start_time = time.time()
+            mem_before = memory_usage(interval=1, timeout=1)
+            df = func(*args, **kwargs)
+            mem_after = memory_usage(interval=1, timeout=1)
+            end_time = time.time()
+            resources = pd.DataFrame(
+                {
+                    "time_s": end_time - start_time,
+                    "memory_gb": (mem_after[0] - mem_before[0]) / 1024,
+                },
+                index=[0],
+            )
+            filename = f"{func.__name__}_resources.csv"
+            output_path = os.path.join(output_dir, filename)
+            resources.to_csv(output_path, index=False)
+            return df
+
+        return wrapper
+
+    return decorator
+
+
+def _parse_dataset_params(
+    request: pytest.FixtureRequest,
+) -> tuple[str | int | Callable[..., pd.DataFrame] | None, ...]:
+    dataset_name = request.config.getoption("--dataset")
+    try:
+        dataset_func = DATASET_GENERATION_FUNCS[dataset_name]
+    except KeyError:
+        raise ValueError(
+            f"{dataset_name} is not a valid dataset. Possible datasets are {','.join(DATASET_GENERATION_FUNCS.keys())}"
+        )
+
+    population = request.config.getoption("--population", default=DEFAULT_POP)
+    try:
+        source = SOURCE_MAPPER[population.lower()]
+    except KeyError:
+        raise ValueError(
+            f"population must be one of USA, RI, or sample. You passed in {population}."
```

Comment:
nit: in all of the docstrings throughout this PR, I'd recommend adding single quotes around the options, e.g. `f"population must be one of 'USA', 'RI', ... in '{population}'."`